Classes/Section Controllers/SearchBar/SearchSegmentBarCell.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  SearchSegementBarCell.swift
+//  Freetime
+//
+//  Created by Ehud Adler on 3/12/19.
+//  Copyright © 2019 Ryan Nystrom. All rights reserved.
+//
+
+import UIKit
+import SnapKit
+
+final class SearchSegmentBarCell: UICollectionViewCell, SearchableCell, UISearchBarDelegate {
+
+    weak var delegate: SearchBarCellDelegate?
+
+    private let searchBar = UISearchBar(frame: .zero)
+    private let segmentControl = UISegmentedControl(frame: .zero)
+    var segmentControlLeadingConstraint: Constraint!
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        backgroundColor = .white
+
+        searchBar.returnKeyType = .search
+        searchBar.enablesReturnKeyAutomatically = false
+        searchBar.searchBarStyle = .minimal
+        searchBar.delegate = self
+
+        contentView.addSubview(searchBar)
+        contentView.addSubview(segmentControl)
+
+        searchBar.snp.makeConstraints { make in
+            make.leading.centerY.equalTo(contentView)
+            make.trailing.equalTo(segmentControl.snp.leading)
+        }
+
+        segmentControl.snp.makeConstraints { make in
+            segmentControlLeadingConstraint = make.leading.equalTo(contentView.snp.trailing).constraint
+            make.trailing.equalTo(contentView.safeAreaLayoutGuide).inset(15)
+            make.top.bottom.equalTo(contentView)
+
+        }
+
+        segmentControl.addTarget(self, action: #selector(updateSegement), for: .valueChanged)
+
+        setSegmentControl()
+        segmentControlLeadingConstraint.deactivate()
```

Comment:
What do you need this constraint for if you `deactivate` it before the `init` returns?

Reply:
Deactivated on start due to the search bar not being in editing mode. Once you start editing the constraint becomes activated